gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/SnappyUpdateDeleteResultSet.java

```diff
@@ -40,7 +40,7 @@ public int modifiedRowCount() {
       rowCount = 0;
       try {
         ExecRow row;
-        while ((row = getNextRow()) != null) {
+        while ((row = getNextRow()) != null && row.nColumns() > 0) {
```

Comment:
@vivekwiz is that also the reason insert-select does not return the count of rows inserted?